### PR TITLE
Getentropy fixes

### PR DESCRIPTION
**Pull request by DesWurstes**
What this pull request does is: 1) It makes it check for func attributes without checking for platforms, which is what other checks do (code style) 2) It doesn't use the backwards-incompatible function silently if the compiler isn't compatible with `weak_import` or it'll create the same problem.

### CODE REVIEW

File: src/random.cpp [hunk 1]
```diff
@@ -28,15 +28,13 @@
 #include <sys/syscall.h>
 #include <linux/random.h>
 #endif
-#if defined(HAVE_GETENTROPY) || (defined(HAVE_GETENTROPY_RAND) && defined(MAC_OSX))
+#if defined(HAVE_GETENTROPY) || (defined(HAVE_GETENTROPY_RAND) && defined(MAC_OSX) && defined(HAVE_FUNC_ATTRIBUTE_WEAK_IMPORT))
 #include <unistd.h>
 #endif
-#if defined(HAVE_GETENTROPY_RAND) && defined(MAC_OSX)
+#if defined(HAVE_GETENTROPY_RAND) && defined(MAC_OSX) && defined(HAVE_FUNC_ATTRIBUTE_WEAK_IMPORT)
```
Comment (jameshilliard) [inline src/random.cpp:34]: I don't think this change is needed.

**Comment by jameshilliard** [comment]
> It doesn't use the backwards-incompatible function silently if the compiler isn't compatible with weak_import or it'll create the same problem.

I don't think this is really any issue, someone using a non-standard incompatible compiler in regards to weak_import isn't likely to be needing backwards compatibility.

**Comment by DesWurstes** [comment]
OK, I thought that was a mistake.

**Comment by jameshilliard** [comment]
I added your style fixes to [#15103](https://github.com/bitcoin/bitcoin/pull/15103).

**Pull request closed without merging**